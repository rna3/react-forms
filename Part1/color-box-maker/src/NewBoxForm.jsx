import React, { useState } from "react";
import { v4 as uuid } from "uuid";

function NewBoxForm({ addBox }) {
  const [formData, setFormData] = useState({ width: "", height: "", color: "" });

  const handleChange = (e) => {
    const { name, value } = e.target;
    setFormData((data) => ({ ...data, [name]: value }));
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    const newBox = { ...formData, id: uuid() };
    addBox(newBox);
    setFormData({ width: "", height: "", color: "" });
  };

  return (
    <form onSubmit={handleSubmit}>
      <label htmlFor="width">Width: </label>
      <input
        id="width"
        name="width"
        type="number"
        value={formData.width}
        onChange={handleChange}
      />
      <label htmlFor="height">Height: </label>
      <input
        id="height"
        name="height"
        type="number"
        value={formData.height}
        onChange={handleChange}
      />
      <label htmlFor="color">Color: </label>
      <input
        id="color"
        name="color"
        type="text"
        value={formData.color}
        onChange={handleChange}
      />
      <button>Add Box</button>
    </form>
  );
}

export default NewBoxForm;
