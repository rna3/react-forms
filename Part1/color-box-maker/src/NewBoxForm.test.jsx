import React from 'react';
import { render } from '@testing-library/react';
import NewBoxForm from './NewBoxForm';

test("renders NewBoxForm without crashing", () => {
    render(<NewBoxForm />);
  });

test("matches snapshot for NewBoxForm", () => {
  const { asFragment } = render(<NewBoxForm />);
  expect(asFragment()).toMatchSnapshot();
});

