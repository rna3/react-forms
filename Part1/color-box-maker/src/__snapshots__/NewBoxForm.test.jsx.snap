// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot for NewBoxForm 1`] = `
<DocumentFragment>
  <form>
    <label
      for="width"
    >
      Width: 
    </label>
    <input
      id="width"
      name="width"
      type="number"
      value=""
    />
    <label
      for="height"
    >
      Height: 
    </label>
    <input
      id="height"
      name="height"
      type="number"
      value=""
    />
    <label
      for="color"
    >
      Color: 
    </label>
    <input
      id="color"
      name="color"
      type="text"
      value=""
    />
    <button>
      Add Box
    </button>
  </form>
</DocumentFragment>
`;
